src/scores/processing/block_bootstrap.py

```diff
@@ -0,0 +1,410 @@
+      Academic press, 2011.
+    """
+
+    objects_list = list(objects)
+
+    # Rename exclude_dims so they are not bootstrapped
+    if exclude_dims is None:
+        exclude_dims = [[] for _ in range(len(objects_list))]
+    if (
+        not isinstance(exclude_dims, list)
+        or len(exclude_dims) != len(objects_list)
+        or not all(isinstance(x, list) for x in exclude_dims)
+    ):
+        raise ValueError(
+            "exclude_dims should be a list of the same length as the number of "
+            "objects containing lists of dimensions to exclude for each object"
+        )
+    renames = []
+    for i, (obj, exclude) in enumerate(zip(objects_list, exclude_dims)):
+        objects_list[i] = obj.rename(
+            {d: f"dim{ii}" for ii, d in enumerate(exclude)},
+        )
+        renames.append({f"dim{ii}": d for ii, d in enumerate(exclude)})
+
+    dim = list(blocks.keys())
+
+    # Ensure bootstrapped dimensions have consistent sizes across objects_list
+    for d in blocks.keys():
+        dim_sizes = [o.sizes[d] for o in objects_list if d in o.dims]
+        if not all(s == dim_sizes[0] for s in dim_sizes):
+            raise ValueError(f"Block dimension {d} is not the same size on all input objects")
+
+    # Retrieve sizes of the bootstrap dimensions from objects_list
+    sizes = None
+    for obj in objects_list:
+        try:
+            sizes = OrderedDict(
+                {d: (obj.sizes[d], b) for d, b in blocks.items()},
+            )
+            break
+        except KeyError:
+            pass
+    if sizes is None:
+        raise ValueError(
+            "At least one input object must contain all dimensions in blocks.keys()",
+        )
+
+    # Generate random indices for bootstrapping all objects_list
+    nested_indices = _n_nested_blocked_random_indices(sizes, n_iteration, circular)
+
+    # Expand indices for broadcasting for each object separately
+    indices = []
+    input_core_dims = []
+    for obj in objects_list:
+        available_dims = [d for d in dim if d in obj.dims]
+        indices_to_expand = [nested_indices[key] for key in available_dims]
+
+        indices.append(_expand_n_nested_random_indices(indices_to_expand))
+        input_core_dims.append(available_dims)
+
+    # Process objects_list separately to handle non-matching dimensions
+    result = []
+    for obj, ind, core_dims in zip(objects_list, indices, input_core_dims):
+        if isinstance(obj, xr.Dataset):
+            # Assume all variables have the same dtype
+            output_dtype = obj[list(obj.data_vars)[0]].dtype
+        else:
+            output_dtype = obj.dtype
+
+        result.append(
+            xr.apply_ufunc(
+                _bootstrap,
+                obj,
+                kwargs={"indices": [ind]},
+                input_core_dims=[core_dims],
+                output_core_dims=[core_dims + ["iteration"]],
+                dask="parallelized",
```

Comment:
Might be worth propagating this to an API argument in `block_bootstrap` so that the user can forcefully disable `dask` for the `ufuncs` for whatever reason.